src/controlbits.rs

```diff
@@ -115,8 +116,8 @@ fn cbrecursion(
     }
     /* B = (p<<16)+c */
 
-    for x in 0..n {
-        temp[x] = (temp[x] << 16) | (x as i32); /* A = (pibar<<16)+id */
+    for (x, itr_temp) in temp.iter_mut().enumerate().take(n) {
+        *itr_temp = (*itr_temp << 16) | (x as i32); /* A = (pibar<<16)+id */
```

Comment:
concerning this specific `take`, if removed it seems to brake the controlbits unit test, however I'm not entirely sure why